### ISSUE TITLE
examples/keras_mnist.py has a very weak scaling from 4 GPUs to 8 GPUs

### ISSUE THREAD

**Issue by denfromufa** [horovod/horovod#998]
**Environment:**
1. Framework: IBM PowerAI 1.6, TF 1.13, Keras 2.1.6:

https://public.dhe.ibm.com/ibmdl/export/pub/software/server/ibm-ai/conda/

2. Framework version: which framework?
3. Horovod version: 0.16.1 (master) compiled with DDL
4. MPI version: OpenMPI 3.1.3, spectrum-mpi | 10.02
5. CUDA version: CUDA toolkit 10.1, GPU driver 418
6. NCCL version: 2.4.2
7. Python version: 3.6
8. OS and version: CentOS 7.5

**Checklist:**
1. Did you search issues to find if somebody asked this question before?
YES
2. If your question is about hang, did you read [this doc](https://github.com/horovod/horovod/blob/master/docs/running.md)?
N/A
3. If your question is about docker, did you read [this doc](https://github.com/horovod/horovod/blob/master/docs/docker.md)?
N/A

**Your question:**

I'm observing very weak scalability for keras_mnist example when scaling from 4 GPUs to 8 GPUs on 1-2 nodes of Power8/P100. Accuracy is quite comparable between different number of GPUs.

40 sec -> 34 sec -> 19sec -> 14 sec
1 GPU -> 2GPU -> 4 GPU -> 8 GPU

![image](https://user-images.githubusercontent.com/7870949/55748892-65287f80-5a05-11e9-9f3f-e723bc755052.png)

Is this something expected?

How to run:

 `bsub -o output84.txt -e error84.txt -gpu "num=4:mode=exclusive_process" -R "span[ptile=4]" -n 8 ddlrun python keras_mnist.py`

P.S.:
Also the loading of data from AWS of the same MNIST dataset by Keras should probably be replaced by np.load('mnist.npz') - I got errors when working with multiple nodes and AWS.

**Comment by alsrgv** [comment]
@denfromufa, MNIST is not the model to test scaling on, it's really just a hello world example.  You don't measure compiler performance by checking how fast it can compile `printf("hello, world!")` :-)

A better test would be to run [TF benchmarks](https://github.com/horovod/horovod/blob/master/docs/benchmarks.md).

**Comment by denfromufa** [comment]
Good to know, let me try this benchmark tomorrow.

I picked the simplest case to make sure that I'm not doing anything wrong.

**Comment by denfromufa** [comment]
Very good job on TF benchmarks up to 64 GPUs, someone probably has spent lots of effort on tuning it :) Because the batch sizes and learning rates are tough tuning parameters to get right for scaling with accuracy over time and is model dependent.

I was able to run the MNIST toy example up to 16 GPUs but for more GPUs it is game over with the accuracy :)

**Comment by alsrgv** [comment]
Indeed, I had similar difficulties with large-batch [Fashion MNIST](https://github.com/horovod/tutorials/blob/master/fashion_mnist/README.md).
